### PR TITLE
Bumping Envoy to 247deb39110eca961af1ee8147a1cbb373c6cced247d

### PR DESCRIPTION
**Pull request by alyssawilk**
https://github.com/envoyproxy/envoy/compare/23e5fc2939e1782416155509edfca323150e8a59...eb39110eca961af1ee8147a1cbb373c6cced247d

### CODE REVIEW

File: library/common/http/client.cc [hunk 1]
```diff
@@ -325,7 +326,9 @@ envoy_stream_intel Client::DirectStreamCallbacks::streamIntel() {
 
 void Client::DirectStream::saveLatestStreamIntel() {
   const auto& info = request_decoder_->streamInfo();
-  stream_intel_.connection_id = info.upstreamConnectionId().value_or(-1);
+  if (info.upstreamInfo()) {
```
Comment (jpsim) [inline library/common/http/client.cc:329]: Do we need to set `connection_id` to `-1` in the case this fails to match previous behavior?

**Review by jpsim — APPROVED** [review]
+1 with one question

File: library/common/http/client.cc (hunk 1 above)
Comment (alyssawilk) [reply library/common/http/client.cc:329]: yeah I added an initializer in the .h file to -1 :-)

**Review by junr03 — APPROVED** [review]
thanks!

**Pull request merged**